lib/service.js

```diff
@@ -280,10 +280,15 @@ AWS.Service = inherit({
    * Gets the signer class for a given request
    * @api private
    */
-  getSignerClass: function getSignerClass() {
+  getSignerClass: function getSignerClass(request) {
     var version;
+    // get operation authtype if present
+    var operation = request ? request.service.api.operations[request.operation] : null;
+    var authtype = operation ? operation.authtype : '';
     if (this.config.signatureVersion) {
       version = this.config.signatureVersion;
+    } else if (authtype === 'v4' || authtype === 'v4-unsigned-body') {
+      version = 'v4';
     } else {
```

Comment:
Cognito Identity Provider uses ["none"](https://github.com/aws/aws-sdk-js/blob/master/apis/cognito-idp-2016-04-18.normal.json#L455) as an authtype; should support for that be added here?